examples/hosts/app-integration/external-data/src/view/debugView.tsx

```diff
@@ -157,15 +140,147 @@ function debugResetExternalData(): void {
 // TODO: Implement simulation of an external data change.  Maybe include UI for the debug user to edit the data
 // themselves (as if they were editing it outside of Fluid).
 // TODO: Consider how we might simulate errors/failures here to play with retry and recovery.
-const ControlsView: React.FC<IControlsViewProps> = (props: IControlsViewProps) => {
-    console.log(props.model);
+const ControlsView: React.FC = () => {
     return (
         <div>
             <h3>Debug controls</h3>
             <div style={{ margin: "10px 0" }}>
                 <button onClick={ debugResetExternalData }>Reset external data</button><br />
-                <button onClick={ props.model.debugSendCustomSignal }>Trigger external data change signal</button><br />
             </div>
         </div>
     );
 };
+
+
+interface ITaskRowProps {
+    task: ExternalDataTask;
+}
+
+/**
+ * The view for a single task in the TaskListView, as a table row.
+ */
+const TaskRow: React.FC<ITaskRowProps> = (props: ITaskRowProps) => {
+    const { task } = props;
+
+    const idChangeHandler = (e: React.SyntheticEvent<HTMLInputElement>): void => {
+        task.id = e.currentTarget.value;
+    };
+    const nameChangeHandler = (e: React.SyntheticEvent<HTMLInputElement>): void => {
+        task.name = e.currentTarget.value;
+    };
+    const priorityChangeHandler = (e: React.SyntheticEvent<HTMLInputElement>): void => {
+        task.priority =  Number.parseInt(e.currentTarget.value, 10);
+    };
+
+    return (
+        <tr>
+            <td><input defaultValue={ task.id }  style={{ width: "30px" }} onChange={ idChangeHandler }></input></td>
+            <td><input defaultValue={ task.name } style={{ width: "200px" }} onChange={ nameChangeHandler }></input></td>
+            <td><input defaultValue={ task.priority } type="number" style={{ width: "50px" }} onChange={ priorityChangeHandler }></input></td>
+        </tr>
+    );
+};
+
+class ExternalDataTask {
+    public id: string;
+    public name: string;
+    public priority: number;
+    public constructor(id: string, name: string, priority: number) {
+        this.id = id;
+        this.name = name;
+        this.priority = priority;
+    }
+}
+/**
+ * A tabular, editable view of the task list.  Includes a save button to sync the changes back to the data source.
+ */
+export const TaskListView: React.FC = () => {
+    const [externalData, setExternalData] = useState({});
+    useEffect(() => {
+        // HACK: Once we have external changes triggering the appropriate Fluid signal, we can simply listen
+        // for changes coming into the model that way.
+        // For now, poll the external service directly for any updates and apply as needed.
+        async function pollForServiceUpdates(): Promise<void> {
```

Comment:
Can we dedupe this logic with ExternalDataView? Maybe pull it out into a helper function they both use?

Reply:
I've pulled it out into a helper function but I'm not sure the helper function should live! Right now I'm leaving it in DebugView.tsx itself, but let's discuss where to move it out to!